perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -435,57 +439,58 @@ sub _remove_service_if_assisted_exists {
-            = $self->problems->search( { external_id => $external_id } )
-            ->first;
-
-        next unless $report;
-
-        # Skip if there is already a report stashed against the service item
-        # name
-        my $service_item_name
-            = $report->get_extra_field_value('service_item_name') // '';
-        next if $missed_collection_reports{$service_item_name};
-
-        my $latest_comment
-            = $report->comments->search(
-                {},
-                { order_by => { -desc => 'id' } },
-        )->first;
-
-        my $report_details = {
-            id          => $report->id,
-            external_id => $report->external_id,
-            open        => $report->is_open,
-            reported    => (
-                $ws->{WorksheetStartDate} eq WHITESPACE_UNDEF_DATE ?
-                '' : $ws->{WorksheetStartDate}
-            ),
-            will_be_completed => (
-                $ws->{WorksheetEscallatedDate} eq WHITESPACE_UNDEF_DATE ?
-                '' : $ws->{WorksheetEscallatedDate}
-            ),
-            latest_comment =>
-                ( $latest_comment ? $latest_comment->text : '' ),
-        };
 
-        $missed_collection_reports{$service_item_name} = $report_details;
+    foreach my $uprn (@uprns) {
+        my $worksheets = $self->whitespace->GetSiteWorksheets($uprn);
+
+        for my $ws (@$worksheets) {
+            next
+                unless $ws->{WorksheetStatusName} eq 'Open'
+                && $ws->{WorksheetSubject} =~ /^Missed/;
+
+            # Check if it exists in our DB
+            my $external_id = 'Whitespace-' . $ws->{WorksheetID};
+            my $report
+                = $self->problems->search( { external_id => $external_id } )
+                ->first;
+
+            next unless $report;
+
+            # Skip if there is already a report stashed against the service item
+            # name
+            my $service_item_name
+                = $report->get_extra_field_value('service_item_name') // '';
+            next if $missed_collection_reports{$service_item_name};
+
+            my $latest_comment
+                = $report->comments->search(
+                    {},
+                    { order_by => { -desc => 'id' } },
+            )->first;
+
+            my $report_details = {
+                id          => $report->id,
+                external_id => $report->external_id,
+                open        => $report->is_open,
+                reported    => (
+                    $ws->{WorksheetStartDate} eq WHITESPACE_UNDEF_DATE ?
+                    '' : $ws->{WorksheetStartDate}
+                ),
+                will_be_completed => (
+                    $ws->{WorksheetEscallatedDate} eq WHITESPACE_UNDEF_DATE ?
+                    '' : $ws->{WorksheetEscallatedDate}
+                ),
+                latest_comment =>
+                    ( $latest_comment ? $latest_comment->text : '' ),
+                uprn => $uprn,
```

Comment:
To just check I understand this - I see this is now looping through both UPRNs if needed, but the UPRN stored here isn't actually used anywhere?

Reply:
Ah yeah that's not needed, have removed in a fixup.